conans/client/graph/graph_builder.py

```diff
@@ -157,12 +160,12 @@ def _prepare_node(node, profile_host, profile_build, down_options):
                     node.conanfile.requires.tool_require(str(tool_require),
                                                          raise_if_duplicated=False)
 
-    def _initialize_requires(self, node, graph, graph_lock):
+    def _initialize_requires(self, node, graph, graph_lock, resolve_prereleases):
```

Comment:
I think we might have another approach here: 
if we define here for all ``requires`` we define that such a ``require`` allows pre-releases, and that is automatically passed to the ``VersionRange`` when requested via property, then the number of modifications necessary might be way less?

Just an idea, I am still not super happy with this PR, it is likely that if we remove the ``=None`` defaults, it would require even more changes.